include/envoy/network/listener.h

```diff
@@ -90,6 +95,12 @@ class ListenerConfig {
    * @return const std::string& the listener's name.
    */
   virtual const std::string& name() const PURE;
+
+  /**
+   * @return factory pointer if listening on UDP socket, otherwise return
+   * nullptr.
+   */
+  virtual Server::ActiveUdpListenerFactory* udpListenerFactory() PURE;
```

Comment:
nit: can this return a const pointer or an optional of a reference_wrapper (I can't remember whether that works or not).